### PR TITLE
Add search product Integration tests

### PR DESCRIPTION
**Pull request by Sampy147**
Added BVA tests:

1. Smallest valid input (1 character) -> success
2. Maximum valid input (100 characters) -> success
3. Just above maximum valid input (e.g 101 characters) -> 400 error
4. One capital letter -> success and should return products that has its smaller case equivalent in its name or description
5. No capital letter -> success and return products that match it in its name or description

These BVA tests were generated in a similar way as the searchProductController unit tests, with some less critical BVA tests left out.

### CODE REVIEW

File: integration-tests/searchProductAPI.test.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  beforeAll(async () => {
+    mongoMemServer = await MongoMemoryServer.create();
+    await mongoose.connect(mongoMemServer.getUri());
+
+    categoryId = new ObjectId("bc7f29ed898fefd6a5f713fd");
+
+    product1 = await productModel.create({
+      name: "First product",
+      slug: "first-product",
+      description: "first product desc " + "a".repeat(100),
```
Comment (jwdavis0200) [inline integration-tests/searchProductAPI.test.js]: mayb we can just do boundary value of a.repeat(101)?

File: integration-tests/searchProductAPI.test.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        `${productPath}/search/${keyword}`
+      );
+
+      expect(response.status).toBe(400);
+      expect(response.body.success).toBe(false);
+      expect(response.body.message).toBe("Keyword is too long");
+    });
+
+    // BVA test: One capital letter that matches a small letter in product name or description
+    it("Should fetch associated products given valid input with one capital letter that matches product name or description, regardless of case", async () => {
```
Comment (jwdavis0200) [inline integration-tests/searchProductAPI.test.js:108]: maybe try with numbers searches in in the input field?

**Review by jwdavis0200 — APPROVED** [review]
LGTM, just minor suggestions if you would be so inclined 💯
